tests/nnstreamer_plugins/unittest_plugins.cc

```diff
@@ -110,6 +110,54 @@
       gst_harness_teardown (h); \
     }
 
+#define GET_MODEL_PATH(model_name) \
+  const gchar *root_path = g_getenv ("NNSTREAMER_BUILD_ROOT_PATH"); \
+  gchar *test_model, *str_launch_line; \
+  \
+  if (root_path == NULL) \
+    root_path = ".."; \
+  test_model = g_build_filename (root_path, "tests", "test_models", "models", \
+    #model_name, NULL); \
+  ASSERT_TRUE (g_file_test (test_model, G_FILE_TEST_EXISTS)); \
+
+/**
+ * @brief Macro for tensor filter auto option test
+ */
+#define TEST_TENSOR_FILTER_AUTO_OPTION(h_desc, fw_name) \
+  GstHarness *h; \
+  gchar *prop_string; \
+ \
+  h = gst_harness_new_empty (); \
+  ASSERT_TRUE (h != NULL); \
+  \
+  gst_harness_add_parse (h, h_desc); \
+  g_free (str_launch_line); \
+  \
+  wait_for_element_state (h->element, GST_STATE_PLAYING); \
+  \
+  gst_harness_get (h, "tensor_filter", "framework", &prop_string, NULL); \
+  EXPECT_TRUE (g_str_equal (prop_string, #fw_name)); \
+  \
+  g_free (prop_string); \
+  gst_harness_teardown (h); \
+  g_free (test_model);
+
+#define wait_for_element_state(element,state) do { \
```

Comment:
Is there any reason applying `do{}while` only for this macro? IMO, applying it all looks better.
FYI, https://stackoverflow.com/questions/257418/do-while-0-what-is-it-good-for

Reply:
Other macros don't need do{}while(0), but it's looks more stable for later variation as you mentioned. Thank you! I will update it.

Reply:
> In case of GET_MODEL_PATH macro, do {} while (0) cannot be used. Some variables such as test_model are treated as local variables inside do-while statement, so following lines cannot use it.
> 
> I updated for TEST_TENSOR_FILTER_AUTO_OPTION macro.

Then, how about refactor GET_MODEL_PATH not to use local variables or defining a static function? 
In addition, this is c++ code so that you can create a class holding such test cases-wide information.

Reply:
Thank you for @helloahn and @wooksong 
I removed the local variable from the macro and applied the do-while(0) statement.